test/jdk/java/net/DatagramSocket/InterruptibleDatagramSocket.java

```diff
@@ -80,16 +80,11 @@ static void test(DatagramSocket s, boolean interruptible) throws Exception {
             if (interruptible)
                 throw e;
             System.out.println("Got expected SocketTimeoutException: " + e);
-        } catch (SocketException e) {
-            if ((e.getCause() instanceof ClosedByInterruptException) && interruptible) {
-                System.out.println("Got expected ClosedByInterruptException: " + e);
-            } else {
+        } catch (SocketException | ClosedByInterruptException e) {
+            if (!interruptible) {
                 throw e;
             }
-        } catch (ClosedByInterruptException e) {
-            if (!interruptible)
-                throw e;
-            System.out.println("Got expected ClosedByInterruptException: " + e);
+            System.out.println("Got expected: " + e);
```

Comment:
Should the check on the exception be a little more focused than blindly accepting any SocketException when the socket is interruptible?

Reply:
+1; the original test accepted `SocketException`s only when the reason was a `ClosedByInterruptException` and `interruptible` was set; since the code never throws a `ClosedByInterruptException`, all `SocketException`s were rejected. After your changes `SocketException`s are accepted.